src/app/dataset-view/additional-components/lineage-component/lineage.component.ts

```diff
@@ -69,62 +56,92 @@ export class LineageComponent extends BaseComponent implements OnInit {
     }
 
     private updateGraph(lineageUpdate: LineageUpdate): void {
-        lineageUpdate.nodes.forEach((dataset: DatasetBasicsFragment) => {
-            this.lineageGraphClusters = this.lineageGraphClusters.map((cluster: ClusterNode) => {
-                if (typeof cluster.childNodeIds === "undefined") {
-                    cluster.childNodeIds = [];
-                }
-
-                if (cluster.label === dataset.kind) {
-                    cluster.childNodeIds.push(dataset.id);
-                }
-                return cluster;
-            });
-        });
-
         const edges = lineageUpdate.edges;
         const currentDataset = lineageUpdate.origin;
-
         this.initLineageGraphProperty();
-
         this.isAvailableLineageGraph = edges.length !== 0;
-
-        const uniqueDatasets: Record<string, DatasetBasicsFragment> = {};
-        edges.forEach((edge: DatasetBasicsFragment[]) =>
-            edge.forEach((dataset: DatasetBasicsFragment) => {
+        const uniqueDatasets: Record<string, DatasetLineageBasics> = {};
+        edges.forEach((edge: DatasetLineageBasics[]) =>
+            edge.forEach((dataset: DatasetLineageBasics) => {
                 uniqueDatasets[dataset.id] = dataset;
             }),
         );
+        this.addSourceGraphNodes(Object.values(uniqueDatasets));
+        this.addDatasetGraphNodes(Object.values(uniqueDatasets), edges, currentDataset);
+    }
 
-        for (const [id, dataset] of Object.entries(uniqueDatasets)) {
+    private sanitizeID(id: string): string {
+        return id.replace(/:/g, "");
+    }
+
+    private getDomainFromUrl(url: string): string {
+        const protocols = ["http", "https", "ftp"];
```

Comment:
Do we even need this list? Can't we just have `new URL(url).hostname` ?

Reply:
Removed this list